File: src/c++/addTwoNumbers.cpp
struct ListNode {
    int val;
    ListNode *next;
    ListNode(int x) : val(x), next(NULL) {}
};

class Solution {
public:
	ListNode * addTwoNumbers(ListNode* l1, ListNode* l2) {
		ListNode head(0),*p=&head;
		int res = 0, sum = 0;
		while (l1 || l2 || res) {
			sum = (l1 ? l1->val:0) + (l2 ? l2->val:0) + res;
			res = 0;
			if (sum > 9) {
				res = 1;
				sum -= 10;
			}
			p->next = new ListNode(sum);
			p = p->next;
			l1 = l1 ? l1->next : l1;
			l2 = l2 ? l2->next : l2;
		}
		return head.next;
	}
};

File: src/python/addTwoNumbers.py
'''
Created on Mar 22, 2018

@author: 44386
'''
'''
ListNode
'''
class ListNode:
    def __init__(self,val):
        self.val=val
        self.next=None
    
def addTwoNumbers(l1, l2):
    idx=ListNode(None)
    n=idx
    res=0
    while(l1.next!=None or l2.next!=None or res==1):
        a=l1.val if l1!=None else 0
        b=l2.val if l2!=None else 0
        sums=a+b+res
        res=0
        if sums>9:
            sums=sums-10
            res=1
        n.next=ListNode(sums)
        n=n.next
        if l1!=None:
            l1=l1.next
        if l2!=None:
            l2=l2.next
    return idx.next


